### PR TITLE
Periodically republish all the block cids we have

### PR DESCRIPTION
**Pull request by ianopolous**

### CODE REVIEW

File: src/main/java/org/peergos/Server.java [hunk 1]
```diff
@@ -117,6 +119,25 @@ public Server(Args args) throws Exception {
...
+                    PeerAddresses ourAddrs = PeerAddresses.fromHost(node);
+                    List<CompletableFuture<Void>> published = refs.stream().parallel()
+                            .map(ref -> dht.provideBlock(ref, node, ourAddrs))
+                            .collect(Collectors.toList());
+                    for (CompletableFuture<Void> fut : published) {
+                        fut.join();
+                    }
+                    long reprovideInterval = 22 * 3600_000L;
+                    Thread.sleep(reprovideInterval);
+                } catch (Exception e) {
```
Comment (kevodwyer) [inline src/main/java/org/peergos/Server.java:135]: make it Throwable to catch Error and Error subclasses as well

**Review by kevodwyer — COMMENTED** [review]
Is provideBlock also called when item first added?

**Comment by ianopolous** [comment]
Provide on add will need to happen separately yes.

File: src/main/java/org/peergos/Server.java (hunk 1 above)
Comment (ianopolous) [reply src/main/java/org/peergos/Server.java:135]: done

**Review by kevodwyer — APPROVED** [review]
Only thought is consider a exponential back-off in case of exception so as not to fill log file in the  case of a network error?

**Comment by ianopolous** [comment]
Backoff where?

**Comment by kevodwyer** [comment]
> Backoff where?

in catch clause. the while loop will contiue immediately

**Pull request merged**